tests/rag_answers/custom_deepeval/metrics/test_factual_correctness.py

```diff
@@ -0,0 +1,580 @@
+                    0.6667,
+                    False,
+                ),  # regular mode, return raw score
+                (False, 0.7, ["fact1", "fact2"], [], 1.0, True),
+                (
+                    True,
+                    0.7,
+                    ["fact1", "fact2", "fact3"],
+                    ["fact4"],
+                    0.0,
+                    False,
+                ),  # strict: score < strict-threshold of 1
+                (
+                    True,
+                    0.7,
+                    ["fact1", "fact2"],
+                    [],
+                    1.0,
+                    True,
+                ),  # strict: score == strict-threshold of 1
+                (
+                    True,
+                    0.7,
+                    ["fact1"],
+                    ["fact2", "fact3"],
+                    0.0,
+                    False,
+                ),  # strict: score <= threshold so should not trigger strict mode logic
+            ],
+        )
+        async def test_strict_mode_can_be_configured(
+            self,
+            strict_mode: bool,
+            threshold: float,
+            TP: list,
+            FP: list,
+            expected_score: float,
+            expected_success: bool,
+            mock_native_model: Mock,
+            test_case: LLMTestCase,
+        ):
+            classified_facts = ClassifiedFacts(TP=TP, FP=FP, FN=[])
+
+            mock_native_model.a_generate = AsyncMock(
+                return_value=(
+                    FactClassificationResult(classified_facts=classified_facts),
+                    0.1,
+                )
+            )
+
+            metric = FactualCorrectnessMetric(
+                model=mock_native_model,
+                threshold=threshold,
+                strict_mode=strict_mode,
+            )
+
+            # let _calculate_score and _finalise_evaluation run naturally
+            result = await metric.a_measure(test_case)
+
+            assert round(result, 4) == round(expected_score, 4)
+            assert metric.success == expected_success
+
+        @pytest.mark.asyncio
+        async def test_native_model_sets_evaluation_costs(
+            self,
+            mock_native_model: Mock,
+            test_case: LLMTestCase,
+            fact_classification_result: FactClassificationResult,
+        ):
+            expected_cost = 0.05
+
+            mock_native_model.a_generate = AsyncMock(
+                return_value=(fact_classification_result, expected_cost)
+            )
+
+            metric = FactualCorrectnessMetric(model=mock_native_model)
+
+            _ = await metric.a_measure(test_case)
+
+            assert metric.evaluation_cost == expected_cost
+            mock_native_model.a_generate.assert_awaited_once()
+
+        @pytest.mark.asyncio
+        @patch(
+            "govuk_chat_evaluation.rag_answers.custom_deepeval.metrics.factual_correctness.factual_correctness.initialize_model"
+        )
```

Comment:
I don't think we need this mock anymore (and that we get a more realistic test without it)